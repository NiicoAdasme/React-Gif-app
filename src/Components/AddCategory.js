import React, { useState } from 'react'
// import GifExpertApp from '../GifExpertApp'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = e => {
        setInputValue(e.target.value);

        // console.log('HandleInputChange llamado')
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        // console.log('HandleSubmit llamado, con valor: ', inputValue);
    
        if(inputValue.trim().length > 2){
            
            setCategories(categ => [inputValue, ...categ]);

            setInputValue('');
        }
    
    };

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input 
                type="text"
                value={inputValue}
                onChange={ handleInputChange }
            />
        </form>  
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};


