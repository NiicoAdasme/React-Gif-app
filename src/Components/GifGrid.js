import React from 'react';
import { useFetchGifs } from '../Hooks/useFetchGifs';
// import { getGifs } from '../Helpers/getGifs';
import { GifGridItem } from './GifGridItem';
// import GifExpertApp from '../GifExpertApp';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

    // const [images, setImages] = useState([]);

    const {data:images, loading} = useFetchGifs(category);

    // getGifs();

    if(category === ""){
        return "Busque una categoria y presione la tecla Enter para buscar el Gif";
    }else{
        return (
            <>
                <h3 className="animate__fadeInDown">Categoria: {category} </h3>
    
                {loading && <p className="animate__flash">Loading...</p>}
    
                <div className="cardGrid">
                {
                    images.map( img => (
                        <GifGridItem 
                            key={img.id}
                            // img={img}
                            {...img} 
                        />
                    ))
                }
            </div>
            </>
            
        );
    }
    
    
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
};