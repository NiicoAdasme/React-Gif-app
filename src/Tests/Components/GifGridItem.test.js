import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {GifGridItem} from '../../Components/GifGridItem';

describe('Pruebas al componente GifGridItem', () => {

    const titulo = 'Un titulo';

    const url = 'https://localhost/algo.jpg';

    const wrapper = shallow(<GifGridItem title={titulo} url={url} />);


    test('Debe retornar el componente', () => {

        expect(wrapper).toMatchSnapshot();

    });
    
    test('Debe de tener una etiqueta <p></p> con el title', () => {
       
        const p = wrapper.find('p').text().trim();

        expect(p).toBe(titulo);


    });
    
    test('Debe de tener la imagen  igual al url y alt de los pros', () => {
       
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(titulo);

    });
    
    test('Debe de tener la clase animate__bounce', () => {
       
        const div = wrapper.find('div');

        const clasName =  div.prop('className').includes('animate__bounce');

        expect(clasName).toBe(true);


    });
    

});
