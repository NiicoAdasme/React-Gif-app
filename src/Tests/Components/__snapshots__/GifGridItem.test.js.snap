// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GifGridItem Debe retornar el componente 1`] = `
<div
  className="card animate__animated animate__bounce animate"
>
  <p>
    Un titulo
  </p>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
</div>
`;
